deployment/ccip/changeset/v1_6/cs_ccip_home.go

```diff
@@ -173,17 +176,60 @@ func validateCommitOffchainConfig(c *pluginconfig.CommitOffchainConfig, selector
 
 func validateUSDCConfig(usdcConfig *pluginconfig.USDCCCTPObserverConfig, state stateview.CCIPOnChainState) error {
 	for sel, token := range usdcConfig.Tokens {
-		onchainState, ok := state.Chains[uint64(sel)]
-		if !ok {
-			return fmt.Errorf("chain %d does not exist in state but provided in USDCCCTPObserverConfig", sel)
-		}
-		if onchainState.USDCTokenPools == nil || onchainState.USDCTokenPools[deployment.Version1_5_1] == nil {
-			return fmt.Errorf("chain %d does not have USDC token pool deployed with version %s", sel, deployment.Version1_5_1)
+		family, err := chain_selectors.GetSelectorFamily(uint64(sel))
+		if err != nil {
+			return fmt.Errorf("failed to find family for selector %d: %w", sel, err)
 		}
-		if common.HexToAddress(token.SourcePoolAddress) != onchainState.USDCTokenPools[deployment.Version1_5_1].Address() {
-			return fmt.Errorf("chain %d has USDC token pool deployed at %s, "+
-				"but SourcePoolAddress %s is provided in USDCCCTPObserverConfig",
-				sel, onchainState.USDCTokenPools[deployment.Version1_5_1].Address().String(), token.SourcePoolAddress)
+		switch family {
+		case chain_selectors.FamilyEVM:
+			onchainState, ok := state.Chains[uint64(sel)]
+			if !ok {
+				return fmt.Errorf("chain %d does not exist in EVM chain state but provided in USDCCCTPObserverConfig", sel)
+			}
+			if onchainState.USDCTokenPools == nil || onchainState.USDCTokenPools[deployment.Version1_5_1] == nil {
+				return fmt.Errorf("chain %d does not have USDC token pool deployed with version %s", sel, deployment.Version1_5_1)
+			}
+			if common.HexToAddress(token.SourcePoolAddress) != onchainState.USDCTokenPools[deployment.Version1_5_1].Address() {
+				return fmt.Errorf("chain %d has USDC token pool deployed at %s, "+
+					"but SourcePoolAddress %s is provided in USDCCCTPObserverConfig",
+					sel, onchainState.USDCTokenPools[deployment.Version1_5_1].Address().String(), token.SourcePoolAddress)
+			}
+		case chain_selectors.FamilySolana:
+			onchainState, ok := state.SolChains[uint64(sel)]
+			if !ok {
+				return fmt.Errorf("chain %d does not exist in Solana chain state but provided in USDCCCTPObserverConfig", sel)
+			}
+			if onchainState.CCTPTokenPool.IsZero() {
+				return fmt.Errorf("chain %d does not have a CCTP token pool deployed", sel)
+			}
+			if onchainState.USDCToken.IsZero() {
+				return fmt.Errorf("chain %d does not have a USDC token in state", sel)
+			}
+			// Calculate the token pool config address for the USDCTokenPool since Solana expects the source pool to be this PDA instead of the actual pool
+			tokenPoolConfig, err := tokens.TokenPoolConfigAddress(onchainState.USDCToken, onchainState.CCTPTokenPool)
+			if err != nil {
+				return fmt.Errorf("failed to calculate token pool config address: %w", err)
+			}
+
+			// Token source pool address is stored as a hex representation of the address bytes
+			// This string is NOT a base58 address
```

Comment:
I think this is the spot where we should check `SourceMessageTransmitterAddr`. Thats the one that needs to be base58 encoded.
```
					// The source pool address when transferring from Solana is really the pool config PDA.
					// Setting this to the string representation of the pool config PDA's bytes to support the following check:
					// https://github.com/smartcontractkit/chainlink-ccip/blob/737dc0373a2a5fd0700aaf786cfa7c3b5d2baa4b/execute/tokendata/usdc/usdc.go#L130
					SourcePoolAddress:            "0xb28f9a6ee1e97193514983a33a2fda122be97e87b1b38946ba6475014da58113",
					SourceMessageTransmitterAddr: "CCitPr8yZbN8zEBEdwju8bnGgKMYcz6XSTbU61CMedj",
```